gbus/worker.go

```diff
@@ -315,68 +315,44 @@ func (worker *worker) processMessage(delivery amqp.Delivery, isRPCreply bool) {
 		return
 	}
 
-	var tx *sql.Tx
-	var txErr error
-	if worker.isTxnl {
-		tx, txErr = worker.txProvider.New()
-		if txErr != nil {
-			worker.log().WithError(txErr).Error("failed to create transaction")
-			worker.span.LogFields(slog.Error(txErr))
-			//reject the message but requeue it so it gets redelivered until we can create transactions
-			_ = worker.reject(true, delivery)
-			return
-		}
-	}
-	err = worker.invokeHandlers(ctx, handlers, bm, &delivery, tx)
-
-	// if all handlers executed with out errors then commit the transactional if the bus is transactional
-	// if the tranaction committed successfully then ack the message.
-	// if the bus is not transactional then just ack the message
+	err = worker.invokeHandlers(ctx, handlers, bm, &delivery)
 	if err == nil {
-		if worker.isTxnl {
-			err = worker.SafeWithRetries(tx.Commit, MaxRetryCount)
-			if err == nil {
-				worker.log().Info("bus transaction committed successfully ")
-				//ack the message
-				_ = worker.ack(delivery)
-			} else {
-				worker.span.LogFields(slog.Error(err))
-				worker.log().WithError(err).Error("failed committing transaction")
-				//if the commit failed we will reject the message
-				_ = worker.reject(false, delivery)
-			}
-		} else { /*if the bus in not transactional just try acking the message*/
-			_ = worker.ack(delivery)
-		}
-		//else there was an error in the invokation then try rollingback the transaction and reject the message
+		_ = worker.ack(delivery)
```

Comment:
Why are you ignoring the error?

Reply:
after examining the code of worker.ack (and worker.nack) the retry strategy defined there retries the action until it succeeds (returns no error) in 100ms intervals

Reply:
then we should fix the function . not to return an error but it would be a whole different PR.

Reply:
yes, will do it in a diff PR